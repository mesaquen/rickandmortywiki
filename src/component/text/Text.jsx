import React, { memo } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const StyledText = styled.span`
  font-size: ${props => props.size};
  font-weight: ${props => (props.bold ? 'bold' : 'normal')};
  color: ${props => props.color};
`;

function Text(props) {
  return <StyledText {...props} />;
}

Text.propTypes = {
  size: PropTypes.string,
  color: PropTypes.string,
  bold: PropTypes.bool,
};

Text.defaultProps = {
  size: '16px',
  color: '#000000',
  bold: false,
};

export default memo(Text);
