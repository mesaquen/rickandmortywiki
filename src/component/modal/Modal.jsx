import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Text from '../text/Text';
import Button from '../button/Button';

const showHideModal = props => (props.visible ? 'block' : 'none');

const Fog = styled.div`
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: 999;
  background: rgba(0, 0, 0, 0.5);
  display: ${showHideModal};
  cursor: pointer;
`;

const Container = styled.div`
  display: flex;
  flex-direction: column;
  background: white;
  position: absolute;
  min-width: 70%;
  min-height: 70%;
  top: 50%;
  left: 50%;
  padding: 20px;
  transform: translate(-50%, -50%);
  cursor: auto;
`;

const Content = styled.div`
  display: flex;
  flex-grow: 1;
`;

const Footer = styled.footer`
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
`;

export default class Modal extends PureComponent {
  static propTypes = {
    children: PropTypes.node,
    visible: PropTypes.bool,
    onClose: PropTypes.func.isRequired,
  };

  static defaultProps = {
    visible: false,
  };

  stopPropagation = event => {
    event.stopPropagation();
  };

  render() {
    const { children, onClose, visible } = this.props;
    return (
      <Fog visible={visible} onClick={onClose}>
        <Container onClick={this.stopPropagation}>
          <Text size="20px" bold color="#444444">
            Character Details
          </Text>
          <Content>{children}</Content>
          <Footer>
            <Button onClick={onClose} flat>
              Close
            </Button>
          </Footer>
        </Container>
      </Fog>
    );
  }
}
