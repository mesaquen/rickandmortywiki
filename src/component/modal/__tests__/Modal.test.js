import React from 'react';
import Renderer from 'react-test-renderer';
import Modal from '../Modal';
import Text from '../../text/Text';
/**
 * @author Mesaque Francisco <mesaquenf@gmail.com>
 * @since 2019-11-16
 */

describe('Modal', () => {
  const callback = jest.fn();
  let component = Renderer.create(<Modal onClose={callback} />);
  it('should have same snapshot when hidden', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  describe('when rendering visible', () => {
    const episodes = [
      { id: 1, episode: 'S01E01', name: 'one' },
      { id: 2, episode: 'S01E02', name: 'two' },
    ];
    beforeEach(() => {
      component = Renderer.create(
        <Modal visible onClose={callback}>
          <Text>Sample</Text>
        </Modal>
      );
    });
    it('should have same snapshot', () => {
      expect(component.toJSON()).toMatchSnapshot();
    });
  });
});
