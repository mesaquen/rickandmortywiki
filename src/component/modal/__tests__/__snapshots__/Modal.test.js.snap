// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should have same snapshot when hidden 1`] = `
<div
  className="sc-htpNat bZypys"
  onClick={[MockFunction]}
>
  <div
    className="sc-bxivhb fEIvGE"
    onClick={[Function]}
  >
    <span
      className="sc-bdVaJa hlPbXB"
      color="#444"
      size="20px"
    >
      Character Details
    </span>
    <div
      className="sc-ifAKCX eSYnoJ"
    />
    <footer
      className="sc-EHOje gFUFTl"
    >
      <button
        className="sc-bwzfXH ksmsEG"
        color="teal"
        data-testid="button"
        onClick={[Function]}
      >
        Close
      </button>
    </footer>
  </div>
</div>
`;

exports[`Modal when rendering visible should have same snapshot 1`] = `
<div
  className="sc-htpNat dOqzDo"
  onClick={[MockFunction]}
>
  <div
    className="sc-bxivhb fEIvGE"
    onClick={[Function]}
  >
    <span
      className="sc-bdVaJa hlPbXB"
      color="#444"
      size="20px"
    >
      Character Details
    </span>
    <div
      className="sc-ifAKCX eSYnoJ"
    >
      <span
        className="sc-bdVaJa lbTbov"
        color="#444"
        size="16px"
      >
        Sample
      </span>
    </div>
    <footer
      className="sc-EHOje gFUFTl"
    >
      <button
        className="sc-bwzfXH ksmsEG"
        color="teal"
        data-testid="button"
        onClick={[Function]}
      >
        Close
      </button>
    </footer>
  </div>
</div>
`;
